platform/darwin/src/http_file_source.mm

```diff
@@ -86,8 +86,18 @@ void cancel() {
 
             NSURLSessionConfiguration *sessionConfig =
             [MGLNetworkConfiguration sharedManager].sessionConfiguration;
+
+            // app might provide a NSURLSessionDelegate for authentication and such
+            id<NSURLSessionDelegate> sessionDelegate = nil;
+            NSDictionary *info = [[NSBundle mainBundle] infoDictionary];
+            NSString *delegateClassName = [info valueForKey:@"MGLNSURLSessionDelegateClassName"];
```

Comment:
Now that `MGLNetworkConfiguration` is public what do you think if instead of reading a class form the Info file we register the class in the network configuration singleton?

- Reading from the plist file is a bit hidden.
- This way it's clear where to look if anyone requires custom authentication.
- It's safer instantiating a class than parsing first a string.